src/emulator/eeprom.c

```diff
@@ -0,0 +1,146 @@
+#include "emulator/eeprom.h"
+#include "emulator/cpu.h"
+#include "emulator/store.h"
+#include "emulator/vc64_RVL.h"
+#include "emulator/xlHeap.h"
+
+_XL_OBJECTTYPE gClassEEPROM = {
+    "MEMORY-PAK",
+    sizeof(EEPROM),
+    NULL,
+    (EventFunc)eepromEvent,
+};
+
+static bool eepromPut8(EEPROM* pEEPROM, u32 nAddress, s8* pData);
+static bool eepromPut16(EEPROM* pEEPROM, u32 nAddress, s16* pData);
+static bool eepromPut32(EEPROM* pEEPROM, u32 nAddress, s32* pData);
+static bool eepromPut64(EEPROM* pEEPROM, u32 nAddress, s64* pData);
+static bool eepromGet8(EEPROM* pEEPROM, u32 nAddress, s8* pData);
+static bool eepromGet16(EEPROM* pEEPROM, u32 nAddress, s16* pData);
+static bool eepromGet32(EEPROM* pEEPROM, u32 nAddress, s32* pData);
+static bool eepromGet64(EEPROM* pEEPROM, u32 nAddress, s64* pData);
+static bool eepromGetBlock(EEPROM* pEEPROM, CpuBlock* pBlock);
+static inline bool eepromEvent_UnknownInline(EEPROM* pEEPROM, void* pArgument);
+
+static bool eepromPut8(EEPROM* pEEPROM, u32 nAddress, s8* pData) { return true; }
+
+static bool eepromPut16(EEPROM* pEEPROM, u32 nAddress, s16* pData) { return true; }
+
+static bool eepromPut32(EEPROM* pEEPROM, u32 nAddress, s32* pData) { return true; }
+
+static bool eepromPut64(EEPROM* pEEPROM, u32 nAddress, s64* pData) { return true; }
+
+static bool eepromGet8(EEPROM* pEEPROM, u32 nAddress, s8* pData) { return true; }
+
+static bool eepromGet16(EEPROM* pEEPROM, u32 nAddress, s16* pData) { return true; }
+
+static bool eepromGet32(EEPROM* pEEPROM, u32 nAddress, s32* pData) { return true; }
+
+static bool eepromGet64(EEPROM* pEEPROM, u32 nAddress, s64* pData) { return true; }
+
+static bool eepromGetBlock(EEPROM* pEEPROM, CpuBlock* pBlock) {
+    if (pBlock->pfUnknown != NULL) {
+        if (!pBlock->pfUnknown(pBlock, 1)) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool fn_80044708(EEPROM* pEEPROM, s32 arg2, u32 nUnknown, void* pBuffer) {
+    SystemRomType eTypeROM;
+    s32 nOffset;
+
+    nOffset = (nUnknown & 0xFFFF) << 5;
+    eTypeROM = gpSystem->eTypeROM;
+
+    switch (eTypeROM) {
+        case 'NKTJ':
+        case 'NKTE':
+        case 'NKTP':
```

Comment:
in general, should we use an enum for this or just write the literals? I'm not sure it matters either way, just want to be consistent in the future

Reply:
I knew I was forgetting something but I couldn't figure out what, now I remember lol

I forgot to make this change, I think using the enum is better because of the description you can get